src/parser/jobs/whm/modules/Lilies.tsx

```diff
@@ -0,0 +1,159 @@
+		onExpiration: this.onLilyGeneration.bind(this),
+		graph: {
+			label: <Trans id="whm.gauge.lily.timer.label">Lily Timer</Trans>,
+			color: GRAPH_DISPLAY_SETTINGS.LILY_GAUGE_COLOR.fade(GRAPH_DISPLAY_SETTINGS.LILY_TIMER_FADE),
+		},
+
+	}))
+
+	private bloodLilyGauge = this.add(new CounterGauge({
+		maximum: LILY_MAX_STACKS,
+		initialValue: 0,
+		graph: {
+			label: <Trans id="whm.gauge.bloodlily.stacks.label">Blood Lily</Trans>,
+			color: GRAPH_DISPLAY_SETTINGS.BLOODLILY_GAUGE_COLOR.fade(GRAPH_DISPLAY_SETTINGS.BLOODLILY_GAUGE_FADE),
+		},
+	}))
+
+	override initialise() {
+		super.initialise()
+
+		const playerFilter = filter<Event>().source(this.parser.actor.id)
+		this.addEventHook(playerFilter.type('action').action(this.data.matchActionId(this.lilyConsumers)), this.onLilySpend)
+		this.addEventHook(playerFilter.type('action').action(this.data.matchActionId(this.bloodLilyConsumers)), () => this.bloodLilyGauge.spend(MISERY_COST))
+		this.addEventHook('complete', this.onComplete)
+		this.lilyTimer.start()
+	}
+
+	private onLilyGeneration() {
+		this.lilyGauge.generate(1)
+		if (!this.lilyGauge.capped) {
+			this.lilyTimer.start()
+		}
+	}
+
+	private onLilySpend() {
+		this.bloodLilyGauge.generate(1)
+		if (this.lilyTimer.expired) {
+			this.lilyTimer.start()
+		}
+		this.lilyGauge.spend(1)
+	}
+
+	private calculateLostLilies(): number {
+		return Math.floor(this.lilyTimer.getExpirationTime() / this.lilyInterval)
+	}
+
+	private onComplete() {
+		const bloodLilyOvercapTiers = {
+			1: SEVERITY.MINOR,
+			2: SEVERITY.MEDIUM,
+			3: SEVERITY.MAJOR,
+		}
+
+		this.suggestions.add(new TieredSuggestion({
+			icon: this.data.actions.AFFLATUS_MISERY.icon,
+			content: <Trans id="whm.gauge.suggestions.bloodlily.overcap.content">
+				Try to use <DataLink action="AFFLATUS_MISERY" /> to avoid wasting Blood Lily growth from overcapping the gauge.
+			</Trans>,
+			tiers: bloodLilyOvercapTiers,
+			value: this.bloodLilyGauge.overCap,
+			why: <Trans id="whm.gauge.suggestions.bloodlily.overcap.why">
+				<Plural value={this.bloodLilyGauge.overCap} one="# Blood lily" other="# Blood lilies" /> did not bloom due to early lily use.
+			</Trans>,
+		})
+		)
+
+		const lilyOvercapSuggestion_600 = <Trans id="whm.gauge.suggestions.lily.overcap.content.600">
+			Try to use <DataLink action="AFFLATUS_RAPTURE" /> or <DataLink action="AFFLATUS_SOLACE" /> before using other GCD heals. It's okay to cap your lilies if you don't need to heal, move, or weave with them.
+		</Trans>
+
+		const lilyOvercapSuggestion_610 = <Trans id="whm.gauge.suggestions.lily.overcap.content.610">
+			Try to use <DataLink action="AFFLATUS_RAPTURE" /> or <DataLink action="AFFLATUS_SOLACE" /> before using other GCD heals. It's okay to overheal with your lilies as they are to be used for mana management and movement aswell as healing.
+		</Trans>
+
+		const lilyOvercapTiers_600 = {1: SEVERITY.MINOR}
+		const lilyOvercapTiers_610 = {
+			1: SEVERITY.MEDIUM,
+			3: SEVERITY.MAJOR,
+		}
```

Comment:
Prefer to define consts at the start of the function instead of inline like this, although severities should be file-level consts.